deployments/extensions.go

```diff
@@ -0,0 +1,405 @@
+package deployments
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v2"
+
+	"github.com/fuseml/fuseml/cli/helpers"
+	"github.com/fuseml/fuseml/cli/kubernetes"
+	"github.com/fuseml/fuseml/cli/paas/ui"
+)
+
+const (
+	defaultDescriptionFileName = "description.yaml"
+	defaultNamespace           = "fuseml-workloads"
+)
+
+type installStep struct {
+	Type      string
+	Location  string
+	Values    string
+	Namespace string
+}
+
+type istioGateway struct {
+	Name        string
+	Port        int
+	ServiceHost string
+}
+
+type extensionDesc struct {
```

Comment:
I think `extensionDesc` is a bit confusing, I would like to suggest `extensionDefinition`.

Reply:
fine

Reply:
although .... it's really called _description_ already in the specs ...

Reply:
I guess that's the good thing about keeping the spec and the implementation on the same PR as there are some things that we can only see on the code and make changes accordingly.

In any case, feel free to ignore my suggestion if you think that `extensionDesc` fits better.

Reply:
Definition could be also used (not just function and scruct names, but also `definition.yaml` file), but I kind of like "description" more...